interceptors/selector/selector.go

```diff
@@ -0,0 +1,58 @@
+package selector
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+type MatchFunc func(ctx context.Context, fullMethod string) bool
+
+type options struct {
+	match MatchFunc
+}
+
+type Option func(o *options)
+
+// WithMatch ...
```

Comment:
I think this comment needs improving
```suggestion
// WithMatch configures what condition should trigger the selector to run the interceptor.
// If used multiple times, the last one specified will be used.
```